projects/idea-hub-smart-contract-solidity/contracts/Authenticated.sol

```diff
@@ -0,0 +1,70 @@
+pragma solidity >=0.4.22 <0.7.0;
+/**
+ * @title Authenticated Contract
+ * @dev Contract feature to allow chosen address to call functions
+*/
+contract Authenticated {
```

Comment:
Could you please add documentation regarding the fields, methods, and arguments? Especially why there are for?